src/js/components/Button/Button.js

```diff
@@ -84,17 +84,30 @@ class Button extends Component {
     } = this.props;
     const { hover } = this.state;
 
-    let buttonIcon = icon;
     // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    let buttonIcon = icon;
+    // nonprimary button icon styling with custom theme
+    if (!primary && icon && !icon.props.color) {
       buttonIcon = cloneElement(icon, {
-        color:
-          theme.global.colors.text[
-            isDarkBackground(this.props) ? 'dark' : 'light'
-          ],
+        color: normalizeColor(theme.button.color, theme),
```

Comment:
What if the theme doesn't have a `button.color`? It seems like we should: `color: normalizeColor(theme.button.color || theme.global.colors.text, theme),`, maybe?